cmd/ci-operator/main_test.go

```diff
@@ -543,3 +543,119 @@ func TestExcludeContextCancelledErrors(t *testing.T) {
 		}
 	}
 }
+
+func TestMultiStageParams(t *testing.T) {
+	testCases := []struct {
+		id             string
+		inputParams    stringSlice
+		expectedParams map[string]string
+		testConfig     []api.TestStepConfiguration
+		expectedErrs   []string
+	}{
+		{
+			id:          "MultiStageTestConfigurationLiteral with valid params",
+			inputParams: stringSlice{[]string{"PARAM1=VAL1", "PARAM2=VAL2"}},
+			testConfig: []api.TestStepConfiguration{
+				{
+					MultiStageTestConfigurationLiteral: &api.MultiStageTestConfigurationLiteral{
+						Environment: map[string]string{
+							"OTHERPARAM": "OTHERVAL",
+						},
+					},
+				},
+			},
+			expectedParams: map[string]string{
+				"PARAM1":     "VAL1",
+				"PARAM2":     "VAL2",
+				"OTHERPARAM": "OTHERVAL",
+			},
+		},
+		{
+			id:          "MultiStageTestConfiguration with valid params",
+			inputParams: stringSlice{[]string{"PARAM1=VAL1", "PARAM2=VAL2"}},
+			testConfig: []api.TestStepConfiguration{
+				{
+					MultiStageTestConfiguration: &api.MultiStageTestConfiguration{
+						Environment: map[string]string{
+							"OTHERPARAM": "OTHERVAL",
+						},
+					},
+				},
+			},
+			expectedParams: map[string]string{
+				"PARAM1":     "VAL1",
+				"PARAM2":     "VAL2",
+				"OTHERPARAM": "OTHERVAL",
+			},
+		},
+		{
+			id:          "Override test param",
+			inputParams: stringSlice{[]string{"PARAM1=NEWVAL", "PARAM2=VAL2"}},
+			testConfig: []api.TestStepConfiguration{
+				{
+					MultiStageTestConfigurationLiteral: &api.MultiStageTestConfigurationLiteral{
+						Environment: map[string]string{
+							"PARAM1": "VAL1",
+						},
+					},
+				},
+			},
+			expectedParams: map[string]string{
+				"PARAM1": "NEWVAL",
+				"PARAM2": "VAL2",
+			},
+		},
+		{
+			id:             "invalid params",
+			inputParams:    stringSlice{[]string{"PARAM1", "PARAM2"}},
+			expectedParams: map[string]string{},
+			expectedErrs: []string{
+				"could not parse multi-stage-param: PARAM1 is not in the format key=value",
+				"could not parse multi-stage-param: PARAM2 is not in the format key=value",
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		configSpec := api.ReleaseBuildConfiguration{
+			Tests: tc.testConfig,
+		}
+
+		o := &options{
+			multiStageParamsSlice: tc.inputParams,
+			configSpec:            &configSpec,
+		}
+
+		errs := overrideMultiStageParams(o)
+		actualParams := make(map[string]string)
+
+		for _, test := range o.configSpec.Tests {
+			if test.MultiStageTestConfigurationLiteral != nil {
+				for name, val := range test.MultiStageTestConfigurationLiteral.Environment {
+					actualParams[name] = val
+				}
+			}
+
+			if test.MultiStageTestConfiguration != nil {
+				for name, val := range test.MultiStageTestConfiguration.Environment {
+					actualParams[name] = val
+				}
+			}
+		}
+		if diff := cmp.Diff(tc.expectedParams, actualParams); diff != "" {
```

Comment:
I think you can cut the implementation some slack and only test for this when there's no error - it is a convention that when `err` is not nil then the other return values can contain garbage, and some implementations take advantage of it, so the outputs only make sense to test when there was no error.